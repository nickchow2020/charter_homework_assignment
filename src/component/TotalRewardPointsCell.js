import React from 'react';

function TotalRewardPointsCell({ buyerId, totalData }) {
  return (
    <tr>
      <th>{buyerId}</th>
      <td>{totalData[buyerId].name}</td>
      <td>{totalData[buyerId].totalPoints}</td>
    </tr>
  );
}

export default TotalRewardPointsCell;
