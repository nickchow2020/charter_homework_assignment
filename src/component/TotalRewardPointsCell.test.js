import { render, screen } from '@testing-library/react';
import TotalRewardPointsCell from './TotalRewardPointsCell';

describe('<TotalRewardPointsCell />', () => {
  const mockData = {
    buyer1: { totalPoints: 165, name: 'test1' },
    buyer2: { totalPoints: 156, name: 'test2' },
    buyer3: { totalPoints: 1325, name: 'test3' },
    buyer4: { totalPoints: 860, name: 'test4' },
  };

  it('should match snapshot', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <TotalRewardPointsCell buyerId={'buyer1'} totalData={mockData} />
        </tbody>
      </table>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain the correct buyerId', () => {
    render(
      <table>
        <tbody>
          <TotalRewardPointsCell buyerId={'buyer1'} totalData={mockData} />
        </tbody>
      </table>
    );

    const buyer1 = screen.getByText('buyer1');

    expect(buyer1).toBeInTheDocument();
  });

  it('should display the correct points', () => {
    render(
      <table>
        <tbody>
          <TotalRewardPointsCell buyerId={'buyer3'} totalData={mockData} />
        </tbody>
      </table>
    );

    const point = screen.getByText('1325');

    expect(point).toBeInTheDocument();
  });

  it('should display buyer name', () => {
    render(
      <table>
        <tbody>
          <TotalRewardPointsCell buyerId={'buyer1'} totalData={mockData} />
        </tbody>
      </table>
    );

    const point = screen.getByText('test1');

    expect(point).toBeInTheDocument();
  });
});
