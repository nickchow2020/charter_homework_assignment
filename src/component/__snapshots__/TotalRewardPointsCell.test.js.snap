// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalRewardPointsCell /> should match snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <th>
          buyer1
        </th>
        <td>
          test1
        </td>
        <td>
          165
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
